tools/event-generator/src/main.rs

```diff
@@ -3,13 +3,13 @@
 
 use proc_macro2::TokenStream;
 use quote::quote;
-use s2n_events::{Output, OutputMode, Result, parser, validation};
+use s2n_events::{Output, OutputConfig, OutputMode, Result, parser, validation};
 
 struct EventInfo<'a> {
     input_path: &'a str,
     output_path: &'a str,
     crate_name: &'a str,
-    output_mode: OutputMode,
+    output_config: OutputConfig,
```

Comment:
Trying to imagine what would be going in output_config in the future. Is output_path going to be in there?

Reply:
My plan for OutputConfig is to put options that will influence the code generation similar to how OutputMode currently does this in [output_mode.rs](https://github.com/aws/s2n-quic/blob/main/tools/s2n-events/src/output_mode.rs). So rather than only influencing the code generation based of ref/mut mode, we can influence it with a [`generate_c_bindings` mode](https://github.com/aws/s2n-quic/commit/17595e2b85064769ddb9361bd1ab2062cc47628e). I wasn't planning on putting stuff like output_path in there, since this is unrelated to the code gen configuration.

I wonder if something like `generate_config` might be a better name maybe 🤔